experimental/GroebnerWalk/Project.toml

```diff
@@ -0,0 +1,8 @@
+name = "GroebnerWalk"
+uuid = "6dc8777d-2b0c-4de5-9e3c-426c1820f6f9"
+authors = ["Kamillo Ferry <kafe@kafe.dev>", "Francesco Nowell <francesconowell@gmail.com>"]
+version = "1.0.1"
```

Comment:
I am not sure we can support "sub-projects" like that.

More importantly, I think code with a license deviating from the license of OSCAR overall is a no-go (at least inside of OSCAR, of course a seperate package based on OSCAR can do anything it wants). Especially a viral one like GPL3 -- including this here would risk "tainting" the whole project with that license... Where does this license choice come from? Is this because your code is based on GPL code written and licensed as such by others and hence must be GPL; or is this your own choice?

Reply:
Regarding the license, we chose GPLv3 because we thought OSCAR is licensed under GPLv3 as well.  This might as well be changed to GPLv3+, that's fine.

Also, regarding "sub-projects", when reworking the code from #3131, I tried to incorporate as many suggestions as possible from there.

That's why we chose to make this an experimental package since 
- then we could develop the code separately from OSCAR and not clutter the main OSCAR module. 
- there is a sizeable set of helper functions which might or might not stay.
- this looked like the appropriate place while we bring the code up to standard.

Ultimately, the goal is to be part of the main OSCAR package and provide functionality similar to `groebner_basis` and its variants. I'm happy to hear about suggestions to make this happen.

Reply:
The things in `experimental/` are not julia packages themselves, but just submodules living in a separate folder. These submodules are loaded like normal Oscar code in `src/`, but it marked in the documentation as "experimental". They don't have a separate license or a separate Project.toml or UUID.

Reply:
Yes my apologies for the confusion about the license, it was ne who was  confused about that for a moment ("brain fart") :oops:. Of course OSCAR is GPL 3 *or later* (but this is not clear in the README so we should clarify it). Still a separate license file is quite unusual, as is a separate Project.toml.

Reply:
> The things in `experimental/` are not julia packages themselves, but just submodules living in a separate folder. These submodules are loaded like normal Oscar code in `src/`, the only difference is that they are marked in the documentation as "experimental" (and things in `src/` should not depend on things `experimental/`). They don't have a separate license or a separate Project.toml or UUID.
> 
> P.S.: This is described in https://docs.oscar-system.org/dev/Experimental/intro/ (but things might not be as clear as we had hoped, so please let us know if there is something that can be improved).

Ah, okay. I understood it that in theory you would just take a package that is to be moved to OSCAR and just plop it down in `experimental/` (and further down the road get the submodule absorbed in OSCAR). It wasn't clear that a separate `Project.toml` is specifically discouraged.